### PR TITLE
refactor(explorer): own Ontology Hub URL state in one module

### LINKED ISSUE #1435: [REFACTOR] Centralize Ontology Hub URL state (ontologyTab / ontologyEntity) in one module
## Description

The `ontologyTab` / `ontologyEntity` URL parameters form an implicit protocol whose readers and writers are scattered across four sites, each with its own `URLSearchParams` handling, bare string literals, and try/catch:

- `explorer/src/App.tsx` — `readInitialWorkspace()` checks both params to open the Ontology Hub workspace
- `explorer/src/workspaces/OntologyWorkspace/index.tsx` — `readTabParam()` maps `ontologyEntity` to the editor tab
- `explorer/src/workspaces/OntologyWorkspace/OntologyEditor.tsx` — `requestedEntityUri()` reads; `selectNode()` writes both params; the ontology dropdown deletes `ontologyEntity`

Renaming a parameter or adding a third one requires touching four files, and the write/clear pairing that fixed the stale-`ontologyEntity` bug in #1278 (review finding 5) is an invariant nothing enforces — the next writer can silently reintroduce it.

## Proposal

Add `explorer/src/workspaces/OntologyWorkspace/ontologyUrlState.ts` owning the protocol:

- `readOntologyUrlState(): { tab?: OntologyHubTab; entityUri?: string }`
- `writeEntitySelection(entityUri: string)` (sets tab=editor + entity together)
- `clearEntitySelection()`
- Parameter names become private constants of this module; all four call sites consume the API.

Pure refactor, no behavior change; the existing node-test suite covers the model layer and can pin the read/write/clear round-trip.

## Acceptance criteria

- No `"ontologyTab"` / `"ontologyEntity"` string literals outside the new module.
- Deep-link open, node selection, ontology switch, and reload behavior unchanged.
- Unit tests for read/write/clear round-trip.

Related: #1278 (review finding 5). I'd like to work on this — please assign to @taoche.

### PR DESCRIPTION
**Pull request by taoche**
## Summary

Implements #1435. The `ontologyTab`/`ontologyEntity` deep-link protocol introduced in #1278 was read and written at **five** scattered sites (the issue counted four; `handleFixInEditor` in `index.tsx` also wrote both params, previously without any try/catch), each with its own `URLSearchParams` handling and bare string literals. The write/clear pairing that fixed the stale-`ontologyEntity` bug in #1278 review was an invariant nothing enforced.

- New `explorer/src/workspaces/OntologyWorkspace/ontologyUrlState.ts` owns the protocol: parameter names are private constants; a pure, directly testable core (`parseOntologyUrlState`/`applyTab`/`applyEntitySelection`/`removeEntitySelection`) under thin DOM shells (`readOntologyUrlState`/`hasOntologyUrlState`/`writeTab`/`writeEntitySelection`/`clearEntitySelection`); the entity/tab pairing and the clear-on-switch rationale are documented at the functions that own them
- All five call sites (`App.tsx`, `index.tsx` ×3, `OntologyEditor.tsx` ×3 spots) converted; a global grep confirms the two literals now exist only in the module and its test
- Pure refactor — behavior identical, including empty-string parameter semantics; the only widening is that `handleFixInEditor`'s previously unguarded `replaceState` now shares the shell's try/catch
- New `explorer/tests/ontologyUrlState.test.ts` (7 tests) pins the read/apply/remove round-trips and tolerance of malformed search strings; wired into `test:graph-workspace`

## Verification

- `npm run test:graph-workspace` 94/94, `tsc -b` clean, eslint identical to baseline (no new findings)
- `pytest tests/explorer/` — 300 passed (1 pre-existing unrelated failure also present on main)

Closes #1435

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

**Review by zeekay — COMMENTED** [review]
Verified this against current `main` (2b866c86). The refactor itself is right and I could not find a behavior difference — it just needs a rebase before it can be merged.

**What I checked**

After resolving the two conflicts locally (details below):

- `npm run test:graph-store` / `test:graph-workspace` / `test:plugin-registry`: 1 / 96 / 7 (89 on `main`, so all 7 of your new cases run — thanks for wiring them into the script rather than leaving them orphaned). `tsc -b` clean. `npm run lint`: 75 problems, byte-identical to `main`.
- Browser check, same four URLs on `main` and on the merged branch, with the registry and ontology endpoints stubbed:
  - `?ontologyEntity=<registered entity>` → Ontology Hub, Editor tab, URL rewritten to `…&ontologyTab=editor`, ontology selected. Identical.
  - `?ontologyEntity=<unregistered nested entity>` → identical (still the #1434 behavior, correctly untouched here).
  - `?ontologyTab=shacl` → identical, including the same pre-existing ErrorBoundary from my incomplete SHACL stub.
  - `?view=graph` → welcome screen. Identical.

The empty-string semantics are the part worth having pinned, and your test does pin it: `params.get()` returning `""` has to stay distinguishable from absent, or `hasOntologyUrlState` stops matching the old `params.has()` and `?ontologyEntity=` would silently stop opening the hub. Splitting the pure `parse`/`apply`/`remove` core from the DOM shells is what makes that testable at all.

**Needs a rebase — both conflicts are mechanical**

1. `explorer/src/App.tsx` — adjacent imports; `main` added two next to yours. Keep all three:

```
import { ExploreWorkspaceTabs, type ExploreView } from './ExploreWorkspaceTabs';
import { fetchAgentMemoryAvailability } from './explorerCapabilities';
import { hasOntologyUrlState } from './workspaces/OntologyWorkspace/ontologyUrlState';
```

2. `explorer/package.json` — `main` now has **two** `"test:graph-workspace"` keys (lines 12 and 13). JSON keeps the last, so the first is dead and the four files only it names have not been running. I ran those four on `main`: 26 tests, all passing. So the clean resolution is one consolidated key — the union, plus your file:

```
"test:graph-workspace": "node --import tsx --test tests/markdownContentViewer.test.ts tests/markdownEditorInteraction.test.tsx tests/markdownEditorState.test.ts tests/nodeMarkdownSync.test.ts tests/graphSceneState.display.test.ts tests/temporalLifecycle.test.ts tests/deterministicExplorerRendering.test.ts tests/explorerCapabilities.test.tsx tests/smallGraphLayout.test.ts tests/realtimeGraphAttributes.test.ts tests/ontologyEditorModel.test.ts tests/ontologyUrlState.test.ts",
```

If you would rather not carry the duplicate-key cleanup in a PR about URL state, appending only your file to the *second* key is enough to make CI run it — but the consolidation is a strict improvement and #1483 is being asked for the same thing, so one of you doing it saves the other a conflict.

**One thing to fix while you are in there**

`updateSearch` writes `?${params.toString()}`, so a URL with a fragment loses it. That is inherited from all five call sites, not introduced here — but this module is now the one place that knows how the URL is written, which makes it the only place the fix belongs. `window.location.hash` appended in `updateSearch` would close it for every caller at once.

Also worth a line in the description: `readOntologyUrlState(search?)` reads live URL state when called with no argument and is pure when given one. That dual role is convenient and the tests only exercise the pure form, so name it in the doc comment as the intended contract rather than leaving it to be discovered.

**Review by Sameer6305 — APPROVED** [review]
Approved.

Reviewed the refactor against #1435, including all existing readers/writers, URL edge cases, deep-link initialization, entity/tab synchronization, ontology switching, reload behavior, React effect ordering, URL encoding, fragment preservation, and the new tests.

I couldn't find a concrete behavioral regression or merge-blocking issue. The Qodo findings around OntologyHubTab typing and protocol literals in tests are reasonable design observations but don't warrant changes in this PR.

The centralization is complete and the added tests meaningfully cover the protocol's wire format and round-trip behavior.

File: explorer/src/workspaces/OntologyWorkspace/ontologyUrlState.ts [hunk 1]
```diff
@@ -0,0 +1,80 @@
+// Sole owner of the Ontology Hub deep-link query parameters: the names below must not be
+// spelled out anywhere else, so that the protocol can change in one place.
+const TAB_PARAM = "ontologyTab";
+const ENTITY_PARAM = "ontologyEntity";
+const EDITOR_TAB = "editor";
+
+export interface OntologyUrlState {
+  /** Raw parameter value; the set of legal tab ids belongs to the workspace, not this module. */
+  tab?: string;
```
[the remark this answers is not in the document]
Comment (Sameer6305) [reply explorer/src/workspaces/OntologyWorkspace/ontologyUrlState.ts:9]: Intentional design choice. The URL-state module keeps the raw tab value and workspace-side validation remains in readInitialTab(). Typing it as OntologyHubTab would require broader dependency/type restructuring, so no change needed here.

File: explorer/tests/ontologyUrlState.test.ts [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+test("writing a tab leaves an existing entity selection alone", () => {
+  const search = applyTab(applyEntitySelection("", "urn:x"), "health");
+  assert.deepEqual(parseOntologyUrlState(search), { tab: "health", entityUri: "urn:x" });
+});
+
+test("absent params read as undefined, blank params as empty strings", () => {
+  assert.deepEqual(parseOntologyUrlState(""), { tab: undefined, entityUri: undefined });
+  assert.deepEqual(parseOntologyUrlState("?other=1"), { tab: undefined, entityUri: undefined });
+  assert.deepEqual(parseOntologyUrlState("?ontologyTab=&ontologyEntity="), {
```
[the remark this answers is not in the document]
Comment (Sameer6305) [reply explorer/tests/ontologyUrlState.test.ts:56]: Intentional. These tests pin the actual URL wire format, so keeping the parameter names as literals ensures a parameter rename is caught rather than making the tests tautological. No change needed.
